strawberry/experimental/pydantic/object_type.py

```diff
@@ -223,24 +225,24 @@ def is_type_of(cls: Type, obj: Any, _info: GraphQLResolveInfo) -> bool:
         if hasattr(cls, "resolve_reference"):
             namespace["resolve_reference"] = cls.resolve_reference
 
-        kwargs: Dict[str, object] = {}
+        dclass_kwargs: DataclassArguments = kwargs or {}
 
         # Python 3.10.1 introduces the kw_only param to `make_dataclass`.
         # If we're on an older version then generate our own custom init function
         # Note: Python 3.10.0 added the `kw_only` param to dataclasses, it was
         # just missed from the `make_dataclass` function:
         # https://github.com/python/cpython/issues/89961
         if sys.version_info >= (3, 10, 1):
-            kwargs["kw_only"] = dataclasses.MISSING
+            dclass_kwargs["kw_only"] = dataclasses.MISSING
```

Comment:
This value here forces me to put it in the TypedDict definition, which in turns make it "wrong" when used to create a dataclass.

I'm unclear on why it's MISSING and not True like the "normal" `type`.